app/src/main/java/dev/arkbuilders/arkmemo/ui/views/SwitchButton.kt

```diff
@@ -0,0 +1,883 @@
+package dev.arkbuilders.arkmemo.ui.views
+
+import android.animation.Animator
+import android.animation.ArgbEvaluator
+import android.animation.ValueAnimator
+import android.animation.ValueAnimator.AnimatorUpdateListener
+import android.content.Context
+import android.content.res.TypedArray
+import android.graphics.Canvas
+import android.graphics.Color
+import android.graphics.Paint
+import android.util.AttributeSet
+import android.view.MotionEvent
+import android.view.View
+import android.widget.Checkable
+import dev.arkbuilders.arkmemo.R
+import dev.arkbuilders.arkmemo.utils.dp2Px
+
+/**
+ * SwitchButton.
+ */
+class SwitchButton : View, Checkable {
+    private val ANIMATE_STATE_NONE = 0
+    private val ANIMATE_STATE_PENDING_DRAG = 1
+    private val ANIMATE_STATE_DRAGING = 2
+    private val ANIMATE_STATE_PENDING_RESET = 3
+    private val ANIMATE_STATE_PENDING_SETTLE = 4
+    private val ANIMATE_STATE_SWITCH = 5
+
+    constructor(context: Context) : super(context) {
+        init(context, null)
+    }
+
+    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
+        init(context, attrs)
+    }
+
+    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(
+        context,
+        attrs,
+        defStyleAttr
+    ) {
+        init(context, attrs)
+    }
+
+    constructor(
+        context: Context,
+        attrs: AttributeSet?,
+        defStyleAttr: Int,
+        defStyleRes: Int
+    ) : super(context, attrs, defStyleAttr, defStyleRes) {
+        init(context, attrs)
+    }
+
+    override fun setPadding(left: Int, top: Int, right: Int, bottom: Int) {
+        super.setPadding(0, 0, 0, 0)
+    }
+
+    private fun init(context: Context, attrs: AttributeSet?) {
+        var typedArray: TypedArray? = null
+        if (attrs != null) {
+            typedArray = context.obtainStyledAttributes(attrs, R.styleable.SwitchButton)
+        }
+        shadowEffect = optBoolean(
+            typedArray,
+            R.styleable.SwitchButton_sb_shadow_effect,
+            true
+        )
+        uncheckCircleColor = optColor(
+            typedArray,
+            R.styleable.SwitchButton_sb_uncheckcircle_color,
+            -0x555556
+        ) //0XffAAAAAA;
```

Comment:
I think we should have a proper comment about the color codes

```suggestion
)
```

Reply:
@shuberttheI I replaced the hard coded int with verbose color code name.